src/components/CustomMessage.tsx

```diff
@@ -75,19 +80,25 @@ export default function CustomMessage(props: Props) {
   }
 
   // Sent by current user
-  if ((message as UserMessage).sender?.userId === userId) {
+  if (
+    message.isUserMessage() &&
+    getSenderUserIdFromMessage(message) === userId
+  ) {
     return (
       <div>
-        {<CurrentUserMessage message={message as UserMessage} />}
-        {activeSpinnerId === message.messageId && (
-          <CustomTypingIndicatorBubble botProfileUrl={botUser?.profileUrl} />
+        {<CurrentUserMessage message={message} />}
+        {activeSpinnerId === message.messageId && botUser && (
+          <CustomTypingIndicatorBubble botProfileUrl={botUser.profileUrl} />
         )}
       </div>
     );
   }
 
   // Sent by other users
-  if (message.isUserMessage() && message.sender?.userId !== botUser.userId) {
+  if (
+    message.isUserMessage() &&
+    getSenderUserIdFromMessage(message) !== (botUser?.userId ?? '##bot_id##')
```

Comment:
Since both the left and right values ​​can be undefined, I have inserted a magic constant for the botUser id.

Reply:
Unlikely but there can be other member with userId "##bot_id##". When can `getSenderUserIdFromMessage(message)` be undefined? I think all messages have sender info unless admin message but admin message is returned before so its ok?

Reply:
make sense! let me remove `##bot_id##`